File: src/main/java/com/nytdacm/oa/util/PasswordUtil.java
package com.nytdacm.oa.util;

import org.apache.commons.codec.digest.DigestUtils;
import org.apache.commons.lang.RandomStringUtils;

public class PasswordUtil {
    public static String getSalt() {
        return RandomStringUtils.random(32, true, true);
    }

    public static String hashPassword(String password, String salt) {
        return DigestUtils.sha256Hex(password + salt);
    }

    public static boolean checkPassword(String password, String salt, String hash) {
        return hashPassword(password, salt).equals(hash);
    }
}

File: oa-utils/src/main/kotlin/com/nytdacm/oa/utils/PasswordUtil.kt
package com.nytdacm.oa.utils

import org.apache.commons.codec.digest.DigestUtils

object PasswordUtil {
    @JvmStatic
    val salt: String
        get() = randomString()

    @JvmStatic
    fun hashPassword(password: String, salt: String): String {
        return DigestUtils.sha256Hex(password + salt)
    }

    @JvmStatic
    fun checkPassword(password: String, salt: String, hash: String): Boolean {
        return hashPassword(password, salt) == hash
    }
}
